### PR TITLE
Pin 842 - Using token id for external communication

### PR DESCRIPTION
**Pull request by beetlecrunch**

### CODE REVIEW

File: src/main/scala/it/pagopa/pdnd/interop/uservice/partymanagement/api/impl/PartyApiServiceImpl.scala [hunk 1]
```diff
@@ -348,49 +352,54 @@ class PartyApiServiceImpl(
...
+  override def consumeToken(tokenId: String, doc: (FileInfo, File))(implicit
     toEntityMarshallerProblem: ToEntityMarshaller[Problem],
     contexts: Seq[(String, String)]
   ): Route = {
 
     val results: Future[Seq[StatusReply[Unit]]] = for {
-      token <- Future.fromTry(Token.decode(token))
+      tokenIdUUID <- tokenId.toFutureUUID
+      token       <- getCommander(tokenId).ask(ref => GetToken(tokenIdUUID, ref))
+      founded     <- token.toFuture(TokenNotFound(tokenId))
```
Comment (ubaldop) [inline src/main/scala/it/pagopa/pdnd/interop/uservice/partymanagement/api/impl/PartyApiServiceImpl.scala]: found?

File: src/main/scala/it/pagopa/pdnd/interop/uservice/partymanagement/api/impl/PartyApiServiceImpl.scala [hunk 2]
```diff
@@ -409,11 +418,13 @@ class PartyApiServiceImpl(
...
   override def invalidateToken(
-    token: String
+    tokenId: String
   )(implicit toEntityMarshallerProblem: ToEntityMarshaller[Problem], contexts: Seq[(String, String)]): Route = {
     val results: Future[Seq[StatusReply[Unit]]] = for {
-      token   <- Future.fromTry(Token.decode(token))
-      results <- processRelationships(token, RejectPartyRelationship)
+      tokenIdUUID <- tokenId.toFutureUUID
+      token       <- getCommander(tokenId).ask(ref => GetToken(tokenIdUUID, ref))
+      founded     <- token.toFuture(TokenNotFound(tokenId))
```
Comment (ubaldop) [inline src/main/scala/it/pagopa/pdnd/interop/uservice/partymanagement/api/impl/PartyApiServiceImpl.scala]: ditto

File: src/main/scala/it/pagopa/pdnd/interop/uservice/partymanagement/model/party/Token.scala [hunk 3]
```diff
@@ -21,11 +19,11 @@ import scala.util.Try
 final case class PartyRelationshipBinding(partyId: UUID, relationshipId: UUID)
 
 final case class Token(
-  id: String,
-  legals: Seq[PartyRelationshipBinding],
-  validity: OffsetDateTime,
+  id: UUID,
   checksum: String,
-  seed: UUID
+  applicationId: String,
```
Comment (ubaldop) [inline src/main/scala/it/pagopa/pdnd/interop/uservice/partymanagement/model/party/Token.scala]: what application is in our domain? Is it the product, a reference to current consumer client id?

**Review by ubaldop — APPROVED** [review]

File: src/main/scala/it/pagopa/pdnd/interop/uservice/partymanagement/model/persistence/serializer/v1/package.scala [hunk 4]
```diff
@@ -20,13 +20,13 @@ package object v1 {
...
-          .traverse[ErrorOr, (String, Token)](ts => getToken(ts.value).map(t => ts.key -> t))
+          .traverse[ErrorOr, (UUID, Token)](ts => getToken(ts.value).map(t => UUID.fromString(ts.key) -> t))
           .map(_.toMap)
         relationships <- state.relationships
-          .traverse[ErrorOr, (String, PersistedPartyRelationship)](rl =>
+          .traverse[ErrorOr, (UUID, PersistedPartyRelationship)](rl =>
             for {
               v <- getPartyRelationship(rl.value)
-            } yield rl.key -> v
+            } yield UUID.fromString(rl.key) -> v
```
Comment (galales) [inline src/main/scala/it/pagopa/pdnd/interop/uservice/partymanagement/model/persistence/serializer/v1/package.scala]: `toUUID` from library

File: src/main/scala/it/pagopa/pdnd/interop/uservice/partymanagement/model/persistence/serializer/v1/package.scala [hunk 5]
```diff
@@ -20,13 +20,13 @@ package object v1 {
           .traverse[ErrorOr, (UUID, Party)](ps => getParty(ps.value).map(p => UUID.fromString(ps.key) -> p))
           .map(_.toMap)
         tokens <- state.tokens
-          .traverse[ErrorOr, (String, Token)](ts => getToken(ts.value).map(t => ts.key -> t))
+          .traverse[ErrorOr, (UUID, Token)](ts => getToken(ts.value).map(t => UUID.fromString(ts.key) -> t))
```
Comment (galales) [inline src/main/scala/it/pagopa/pdnd/interop/uservice/partymanagement/model/persistence/serializer/v1/package.scala]: ditto `toUUID`

File: src/main/scala/it/pagopa/pdnd/interop/uservice/partymanagement/model/party/Token.scala [hunk 6]
```diff
@@ -49,23 +47,13 @@ object Token extends SprayJsonSupport with DefaultJsonProtocol {
       .find(_.role == Manager)
       .map(managerRelationship =>
         Token(
-          id = managerRelationship.applicationId,
-          seed = UUID.fromString(tokenSeed.seed),
+          id = UUID.fromString(tokenSeed.id),
```
Comment (galales) [inline src/main/scala/it/pagopa/pdnd/interop/uservice/partymanagement/model/party/Token.scala]: ditto2 `toUUID`

File: src/main/protobuf/v1/token.proto [hunk 7]
```diff
@@ -10,7 +10,7 @@ message TokenV1 {
   required string id = 1;
   repeated PartyRelationshipBindingV1 legals = 2;
   required string validity = 3;
-  required string seed = 4;
+  required string applicationId = 4;
```
Comment (galales) [inline src/main/protobuf/v1/token.proto]: what is the `applicationId`?

**Review by galales — APPROVED** [review]

File: src/main/scala/it/pagopa/pdnd/interop/uservice/partymanagement/api/impl/PartyApiServiceImpl.scala (hunk 1 above)
Comment (beetlecrunch) [reply src/main/scala/it/pagopa/pdnd/interop/uservice/partymanagement/api/impl/PartyApiServiceImpl.scala]: I'll fix it.

File: src/main/scala/it/pagopa/pdnd/interop/uservice/partymanagement/api/impl/PartyApiServiceImpl.scala (hunk 2 above)
Comment (beetlecrunch) [reply src/main/scala/it/pagopa/pdnd/interop/uservice/partymanagement/api/impl/PartyApiServiceImpl.scala]: I'll fix it.

File: src/main/scala/it/pagopa/pdnd/interop/uservice/partymanagement/model/party/Token.scala (hunk 3 above)
Comment (beetlecrunch) [reply src/main/scala/it/pagopa/pdnd/interop/uservice/partymanagement/model/party/Token.scala]: [Here](https://github.com/pagopa/pdnd-interop-uservice-party-management/blob/2004dd839ec9d922ecbd098b02a0671e017998af/src/main/scala/it/pagopa/pdnd/interop/uservice/partymanagement/model/party/PersistedPartyRelationship.scala#L25) the explanation of what is `applicationId`.

I think it is not necessary.

File: src/main/protobuf/v1/token.proto (hunk 7 above)
Comment (beetlecrunch) [reply src/main/protobuf/v1/token.proto]: See [comment](https://github.com/pagopa/pdnd-interop-uservice-party-management/pull/36#discussion_r761249866).

File: src/main/scala/it/pagopa/pdnd/interop/uservice/partymanagement/model/party/Token.scala (hunk 6 above)
Comment (galales) [reply src/main/scala/it/pagopa/pdnd/interop/uservice/partymanagement/model/party/Token.scala]: _don't forget about me_ 🥲

File: kubernetes/deployment.yaml.template [hunk 8]
```diff
@@ -86,8 +86,8 @@ spec:
                 secretKeyRef:
                   name: storage
                   key: STORAGE_PSW
-            - name: TOKEN_VALIDITY
-              value: {{tokenValidity}}
+            - name: $TOKEN_VALIDITY_HOURS
```
Comment (ubaldop) [inline kubernetes/deployment.yaml.template]: why are you using the dollar sign here?

File: src/main/scala/it/pagopa/pdnd/interop/uservice/partymanagement/common/system/ApplicationConfiguration.scala [hunk 9]
```diff
@@ -15,7 +15,7 @@ object ApplicationConfiguration {
     config.getString("uservice-party-management.storage.type")
   }
 
-  def tokenValidity: Long = {
-    config.getLong("uservice-party-management.token-validity")
+  def tokenValidityHours: Long = {
```
Comment (ubaldop) [inline src/main/scala/it/pagopa/pdnd/interop/uservice/partymanagement/common/system/ApplicationConfiguration.scala:18]: saying hours are we assuming that the env var must define hours, right? What if applications need a less than one hour validity though?

Sorry for the silly question, maybe I am missing some specs 😊

**Review by ubaldop — APPROVED** [review]

File: src/main/scala/it/pagopa/pdnd/interop/uservice/partymanagement/common/system/ApplicationConfiguration.scala (hunk 9 above)
Comment (beetlecrunch) [reply src/main/scala/it/pagopa/pdnd/interop/uservice/partymanagement/common/system/ApplicationConfiguration.scala:18]: At this moment, there are no specs about that.
I would extract this value from the code.

**Pull request merged**